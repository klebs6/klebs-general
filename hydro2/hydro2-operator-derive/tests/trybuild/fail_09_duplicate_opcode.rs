// ---------------- [ File: hydro2-operator-derive/tests/trybuild/fail_09_duplicate_opcode.rs ]
//! Specifying `opcode` twice â†’ compile_fail.
use hydro2_operator_derive::*;
use hydro2_operator::*; 
use named_item_derive::*;
use named_item::*;

#[derive(Debug,NamedItem,Operator)]
#[operator(
    execute="example_fn",
    opcode="BasicOpCode::TestOp",
    opcode="BasicOpCode::AnotherOp" // should fail
)]
pub struct DuplicateOpcode {
    name: String,
}

impl DuplicateOpcode {
    pub async fn example_fn(&self) -> NetResult<()> {
        Ok(())
    }
}

fn main() {}

