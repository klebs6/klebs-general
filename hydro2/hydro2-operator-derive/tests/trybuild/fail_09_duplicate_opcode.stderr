error: Opcode key was provided more than once in `#[operator(...)]`.
  --> tests/trybuild/fail_09_duplicate_opcode.rs:12:5
   |
12 |     opcode="OpCode::AnotherOp" // should fail
   |     ^^^^^^
